### PR TITLE
Adds Error label to inputs

### PR DESCRIPTION
**Pull request by Mera-Gangapersaud**
# Description 

#1274 was closed by accident. This is the same branch. The Moments Page now displays an error message below the input if it's left empty.

## More Details

I added an Error Label Component for Inputs so the error message shows up below the textInputArea.

## Corresponding Issue

#1085 

# Screenshots

screenshot from 2018-11-22 09-45-40

# Test Coverage
✅

### CODE REVIEW

File: client/app/components/Input/InputError.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { I18n } from '../../libs/i18n';
+
+export type Props = {
+  error?: boolean,
+};
+
+export const InputError = (props: Props) => {
+  const { error } = props;
+  return (
+    <div className={css.labelInfo}>
```
Comment (julianguyen) [inline client/app/components/Input/InputError.jsx]: Let's do this instead:
```
if (!error) return null;
return (
<div className={css.labelInfo}>
	<div className={`labelError ${globalCss.smallMarginTop} ${css.label}`}>
		<FontAwesomeIcon icon={faExclamation} />
		&emsp;
		{I18n.t('common.form.error_explanation')}
	</div>
</div>
);
```

File: client/app/components/Input/InputError.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// @flow
```
Comment (julianguyen) [inline client/app/components/Input/InputError.jsx]: Let's write a unit test for this, `client/app/components/Input/__tests__/InputLabel.spec.jsx` is a good example to follow.

File: client/app/components/Input/index.jsx [hunk 3]
```diff
@@ -236,6 +237,15 @@ export class Input extends React.Component<Props, State> {
     return null;
   };
 
+  displayError = () => {
+    const { type } = this.props;
+    const { error } = this.state;
+    if (REQUIRES_LABEL.includes(type)) {
```
Comment (julianguyen) [inline client/app/components/Input/index.jsx]: I think we always want to display an error even if the component may not have a label.

**Comment by julianguyen** [comment]
Any updates on this? :)

**Comment by geppy** [comment]
@julianguyen Could we make a specific error message for each input?  `common.form.error_explanation` applies to the form as a whole.

**Comment by julianguyen** [comment]
We could @geppy, but I think there's only one type of error i.e. empty value. Are we missing anything else though?

**Comment by julianguyen** [comment]
@geppy That's a good point, and they should come from the server! Would you be interested in investigating this further? Thanks for bringing it up!

**Comment by julianguyen** [comment]
Hey @Mera-Gangapersaud any updates on this?

**Pull request closed without merging**